src/components/Article.jsx

```diff
@@ -1,21 +1,40 @@
 import { useState, useEffect } from "react";
-import { getArticle, getComments } from "../api";
-import { useLocation } from "react-router-dom";
+import { getArticle, getComments, updateVotes } from "../api";
+import { useParams } from "react-router-dom";
 import CommentCard from "./CommentCard";
+import ThumbUpOutlinedIcon from "@mui/icons-material/ThumbUpOutlined";
+import ThumbDownOutlinedIcon from "@mui/icons-material/ThumbDownOutlined";
 
 const Article = (props) => {
   const [fullArticle, setFullArticle] = useState({});
   const [comments, setComments] = useState([]);
-  const location = useLocation();
-  const { article_id } = location.state;
+  const [vote, setVote] = useState(0);
+  const [voteError, setVoteError] = useState("");
+  const { article_id } = useParams();
 
   useEffect(() => {
-    getArticle(article_id).then((result) => setFullArticle(result.article));
-  }, []);
+    getArticle(article_id).then((result) => {
+      setFullArticle(result.article);
+      setVote(result.article.votes);
+    });
+  }, [article_id]);
+
   useEffect(() => {
     getComments(article_id).then((result) => setComments(result.comments));
   }, []);
 
+  const handleUpVote = () => {
```

Comment:
Two quite similar functions, could they be refactored into one function which takes an argument? 🤔